azure/scope/cluster.go

```diff
@@ -266,20 +267,19 @@ func (s *ClusterScope) NSGSpecs() []azure.NSGSpec {
 }
 
 // SubnetSpecs returns the subnets specs.
-func (s *ClusterScope) SubnetSpecs() []azure.SubnetSpec {
-	numberOfSubnets := len(s.AzureCluster.Spec.NetworkSpec.Subnets)
-	if s.AzureCluster.Spec.BastionSpec.AzureBastion != nil {
-		numberOfSubnets++
-	}
+func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
+	subnetSpecs := make([]azure.ResourceSpecGetter, 0, len(s.AzureCluster.Spec.NetworkSpec.Subnets)+1)
```

Comment:
it should be `len(s.AzureCluster.Spec.NetworkSpec.Subnets)+1` only if `s.AzureCluster.Spec.BastionSpec.AzureBastion != nil`, otherwise the slice should have capacity `len(s.AzureCluster.Spec.NetworkSpec.Subnets)`

what was there before seemed correct, why change it?